migrations/v5-MIGRATION.md

```diff
@@ -0,0 +1,75 @@
+| `identifyWith(String, (PurchasesError) -> Unit, (PurchaserInfo) -> Unit)` |
+| `resetWith((PurchasesError) -> Unit, (PurchaserInfo) -> Unit)` |
+
+| Deprecated | New  |
+|------------|------|
+| `configure(Context, String, String?, Boolean, ExecutorService)` | `configure(PurchasesConfiguration)` |
+| `invalidatePurchaserInfoCache()` | `invalidateCustomerInfoCache()` |
+| `removeUpdatedPurchaserInfoListener()` | `removeUpdatedCustomerInfoListener()` |
+| `getPurchaserInfo(ReceivePurchaserInfoListener)` | `getCustomerInfo(ReceiveCustomerInfoCallback)` |
+| `restorePurchases(ReceivePurchaserInfoListener)` | `restorePurchases(ReceiveCustomerInfoCallback)` |
+| `logOut(ReceivePurchaserInfoListener)` | `logOut(ReceiveCustomerInfoCallback)` |
+| `purchaseProduct(Activity, SkuDetails, MakePurchaseListener)` | `purchaseProduct(Activity, StoreProduct, PurchaseCallback)` |
+| `purchaseProduct(Activity, SkuDetails, UpgradeInfo, ProductChangeListener)` | `purchaseProduct(Activity, StoreProduct, UpgradeInfo, ProductChangeCallback)` |
+| `purchasePackage(Activity, Package, MakePurchaseListener)` | `purchasePackage(Activity, Package, PurchaseCallback)` |
+| `purchasePackage(Activity, Package, UpgradeInfo, ProductChangeListener)` | `purchasePackage(Activity, Package, UpgradeInfo, ProductChangeCallback)` |
+
+## Kotlin Helpers Changes
+
+| Old signature | New signature |
+|---------------|---------------|
+| `getPurchaserInfoWith((PurchasesError) -> Unit, (PurchaserInfo) -> Unit)` | `getCustomerInfoWith((PurchasesError) -> Unit, (CustomerInfo) -> Unit)` |
+| `purchasePackageWith(Activity, Package, (PurchasesError) -> Unit, (Purchase, PurchaserInfo) -> Unit)` | `purchasePackage(Activity, Package, (PurchasesError) -> Unit, (StoreTransaction, CustomerInfo) -> Unit)` |
+| `purchasePackageWith(Activity, Package, UpgradeInfo, (PurchasesError) -> Unit, (Purchase, PurchaserInfo) -> Unit)` | `purchasePackage(Activity, Package, UpgradeInfo, (PurchasesError) -> Unit, (StoreTransaction, CustomerInfo) -> Unit)` |
+| `purchaseProductWith(Activity, SkuDetails, (PurchasesError) -> Unit, (Purchase, PurchaserInfo) -> Unit)` | `purchaseProductWith(Activity, StoreProduct, (PurchasesError) -> Unit, (StoreTransaction, CustomerInfo) -> Unit)` |
+| `purchaseProductWith(Activity, SkuDetails, UpgradeInfo, (PurchasesError) -> Unit, (Purchase, PurchaserInfo) -> Unit)` | `purchaseProductWith(Activity, StoreProduct, UpgradeInfo, (PurchasesError) -> Unit, (StoreTransaction, CustomerInfo) -> Unit)` |
+
+## Type Changes
+
+| Old type name | New type name |
+|------------|------|
+| `PurchaserInfo` | `CustomerInfo` |
+| `ReceivePurchaserInfoListener` | `ReceiveCustomerInfoCallback` |
+| `UpdatedPurchaserInfoListener` | `UpdatedCustomerInfoListener` |
+| `ReceiveOfferingsListener` | `ReceiveOfferingsCallback` |
+| `PurchasesErrorListener` | `PurchasesErrorCallback` |
+
+- `PurchaserInfo` has been deprecated and renamed to `CustomerInfo`. This rename also affects to all functions that had 
```

Comment:
from here on is just describing the changes in the tables above, right? maybe we can just have little "intro" sections at the top of each section above, kinda like how our [ios doc](https://github.com/RevenueCat/purchases-ios/blob/main/docs/V4_API_Updates.md#new-types) does it